### PR TITLE
Create enumerable

### PR DESCRIPTION
**Pull request by tafaramafemba**
### Hello

In this project we managed to do the following:

1. Created a class MyList that has an instance variable @list.
2. In MyList implemented a method #each that yields successive members of @list and uses the MyEnumerable module.
3. Created a module MyEnumerable that implements the following methods :
#all?
#any?
#filter
#max (optional)
#min (optional)
#sort (optional)
4. Each class and module has a separate .rb file.

### CODE REVIEW

File: my_enumerable.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
+module MyEnumerable
+  def all?
+    each do |element|
```
Comment (zainsadaqat) [inline my_enumerable.rb:3]: You required an array to run a loop. Please fix the syntax 👍🏻

File: my_enumerable.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      return false unless yield(element)
+    end
+    true
+  end
+
+  def any?
+    each do |element|
```
Comment (zainsadaqat) [inline my_enumerable.rb:10]: You required an array to run a loop. Please fix the syntax 👍🏻

File: my_enumerable.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+      return true if yield(element)
+    end
+    false
+  end
+
+  def filter
+    result = []
+    each do |element|
```
Comment (zainsadaqat) [inline my_enumerable.rb:18]: You required an array to run a loop. Please fix the syntax 👍🏻

File: my_enumerable.rb [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    max = nil
+    each do |element|
+      max = element if max.nil? || element > max
+    end
+    max
+  end
+
+  def min
+    min = nil
+    each do |element|
```
Comment (zainsadaqat) [inline my_enumerable.rb:34]: You required an array to run a loop. Please fix the syntax 👍🏻

File: my_enumerable.rb [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+      min = element if min.nil? || element < min
+    end
+    min
+  end
+
+  def sort
+    result = []
+    each do |element|
```
Comment (zainsadaqat) [inline my_enumerable.rb:42]: You required an array to run a loop. Please fix the syntax 👍🏻

**Review by zainsadaqat — CHANGES_REQUESTED** [review]
# Required Changes 🔁
# Hello @tafaramafemba and @jsug9  😄 

You're doing a great job on your project. We're proud of you. Good job so far! 

There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
  ✅  Correct use of Github Flow
  ✅  No linter errors
  ✅  Descriptive Pull Request

## Required Changes ♻️

_Check the comments under the review._




## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding! 👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by jsug9** [comment]
Dear @zainsadaqat I don't understand the required changes, these functions are looping throught each of the elements in the array. Could you give us a little explanation?

**Comment by jsug9** [comment]
Captura de Pantalla 2022-06-09 a la(s) 2 32 32 p  m
Dear @zainsadaqat, here I attach the output of my code.

**Comment by zainsadaqat** [comment]
> Captura de Pantalla 2022-06-09 a la(s) 2 32 32 p  m
> 
> Dear @zainsadaqat, here I attach the output of my code.


-- 

### Alright, You're good to go 👍🏻

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @tafaramafemba and @jsug9,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**